File: android/src/main/kotlin/com/iblurblur/flutter_hero_plugin/ColorStreamHandler.kt
package com.iblurblur.flutter_hero_plugin

import android.os.Handler
import android.os.Looper
import io.flutter.plugin.common.EventChannel

class ColorStreamHandler : EventChannel.StreamHandler {
    private var eventSink: EventChannel.EventSink? = null
    private val colors = listOf(0xff71C9CE, 0xffA6E3E9, 0xffCBF1F5, 0xffE3FDFD)
    private var index = 0

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
        changeColor()
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }

    private fun changeColor() {
        val handler = Handler(Looper.getMainLooper())
        handler.post(object : Runnable {
            override fun run() {
                if (index >= colors.count()) {
                    index = 0
                }
                eventSink?.success(colors[index])
                index += 1
                handler.postDelayed(this, 3000)
            }
        })
    }
}

File: ios/Classes/ColorStreamHandler.swift
import Foundation

class ColorStreamHandler:NSObject, FlutterStreamHandler {
    
    private var eventSink: FlutterEventSink?
    private let colors = [0xff71C9CE, 0xffA6E3E9, 0xffCBF1F5, 0xffE3FDFD]
    private var index = 0
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        changeColor()
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        self.eventSink = nil
        return nil
    }
    
    func changeColor()  {
        Timer.scheduledTimer(timeInterval: 3, target: self, selector: #selector(colorEvent), userInfo: nil, repeats: true)
    }
    
    @objc func colorEvent(){
        guard let eventSink = eventSink else {
            return
        }
        
        if index >= colors.count {
            index = 0
        }
        eventSink(colors[index])
        index += 1
    }
}
